src/Symfony/Component/Cache/Adapter/PdoAdapter.php

```diff
@@ -360,30 +357,32 @@ private function getConnection(): \PDO
             $this->conn = new \PDO($this->dsn, $this->username, $this->password, $this->connectionOptions);
             $this->conn->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
         }
-        $this->driver ??= $this->conn->getAttribute(\PDO::ATTR_DRIVER_NAME);
 
         return $this->conn;
     }
 
+    private function getDriver(): string
+    {
+        return $this->driver ??= $this->getConnection()->getAttribute(\PDO::ATTR_DRIVER_NAME);
+    }
+
     private function getServerVersion(): string
     {
-        return $this->serverVersion ??= $this->conn->getAttribute(\PDO::ATTR_SERVER_VERSION);
+        return $this->serverVersion ??= $this->getConnection()->getAttribute(\PDO::ATTR_SERVER_VERSION);
     }
 
     private function isTableMissing(\PDOException $exception): bool
     {
-        $driver = $this->driver;
+        $driver = $this->getDriver();
         $code = $exception->getCode();
 
-        switch (true) {
-            case 'pgsql' === $driver && '42P01' === $code:
-            case 'sqlite' === $driver && str_contains($exception->getMessage(), 'no such table:'):
-            case 'oci' === $driver && 942 === $code:
-            case 'sqlsrv' === $driver && 208 === $code:
-            case 'mysql' === $driver && 1146 === $code:
-                return true;
-            default:
-                return false;
-        }
+        return match (true) {
+            'pgsql' === $driver && '42P01' === $code,
+            'sqlite' === $driver && str_contains($exception->getMessage(), 'no such table:'),
+            'oci' === $driver && 942 === $code,
+            'sqlsrv' === $driver && 208 === $code,
+            'mysql' === $driver && 1146 === $code => true,
+            default => false,
+        };
```

Comment:
```suggestion
        return match ($driver) {
            'pgsql' => '42P01' === $code,
            'sqlite' => str_contains($exception->getMessage(), 'no such table:'),
            'oci' => 942 === $code,
            'sqlsrv' => 208 === $code,
            'mysql' => 1146 === $code,
            default => false,
        };
```

Reply:
Good point, looks much better this way, done.